backend/api/v1/sql_service.go

```diff
@@ -1331,8 +1339,16 @@ func (s *SQLService) checkQueryRights(
 		databaseMessageMap[database] = databaseMessage
 	}
 
+	if len(databaseMessageMap) == 0 && project == nil {
+		project, _, err = s.getProjectAndDatabaseMessage(ctx, instance, databaseName)
+		if err != nil {
+			return err
+		}
+	}
+
 	if project == nil {
-		return status.Error(codes.NotFound, "project not found")
+		// Never happen
+		return status.Error(codes.Internal, "[UNEXPECTED] project not found")
```

Comment:
Don't need to put Unexpected in the string. Internal error status code is unexpected.

Same above.